### DISCUSSION TITLE
Feature request: getting excel worksheet names and index only

### DISCUSSION THREAD

**Discussion by 13minutes-yt** [dathere/qsv#1697] [Ideas]
getting `--metadata c `to be performed on an excel file takes some time for huge excel files. Is it possible to get the sheet name and the index only thereby cutting the time to process all the rows and columns( like an argument -a or --metadata a). I am currently using `csvtk xlsx2csv input.xlsx -a` which is much faster than qsv since it gets only sheetname and index. (Note: qsv is blazingly faster in outputting csv :) ).

**Comment by jqnatividad** [comment]
This is a good request that can be quickly implemented. Didn't realize getting all that metadata took time...

Created #1698

**Reply by 13minutes-yt** [reply]
I compiled it from source. still slow. is there something missing?

**Reply by jqnatividad** [reply]
@13minutes-yt I was still fetching range data... that must be the slowdown. Fixed it
https://github.com/jqnatividad/qsv/commit/09dd54bcab3c077878f1fc847bcf34c0ad2ff138

Can you give it a try and confirm?

**Reply by 13minutes-yt** [reply]
All good. time dropped from `10.7 sec `to` 0.16 sec` on my **57mb** xlsx file with 330,000 rows. 😄

**Reply by jqnatividad** [reply]
Tweaked it a bit more to return sheet type and visible info - already got the info in the sheet struct, so might as well...

https://github.com/jqnatividad/qsv/commit/8588e1571d22d8048ad3a9649b282bc12b3db81a
